client/signup/steps/user/index.jsx

```diff
@@ -200,16 +205,24 @@ export class UserStep extends Component {
 
 		this.props.recordTracksEvent( 'calypso_signup_user_step_submit', analyticsData );
 
-		recordGoogleRecaptchaAction( this.state.recaptchaClientId, 'calypso/signup/formSubmit' ).then(
-			token => {
-				this.submit( {
-					userData,
-					form: formWithoutPassword,
-					queryArgs: ( this.props.initialContext && this.props.initialContext.query ) || {},
-					recaptchaToken: token,
-				} );
-			}
-		);
+		if ( 'show' === abtest( 'userStepRecaptcha' ) ) {
+			recordGoogleRecaptchaAction( this.state.recaptchaClientId, 'calypso/signup/formSubmit' ).then(
+				token => {
+					this.submit( {
+						userData,
+						form: formWithoutPassword,
+						queryArgs: ( this.props.initialContext && this.props.initialContext.query ) || {},
+						recaptchaToken: token,
+					} );
+				}
+			);
+		} else {
+			this.submit( {
+				userData,
+				form: formWithoutPassword,
+				queryArgs: ( this.props.initialContext && this.props.initialContext.query ) || {},
+			} );
+		}
```

Comment:
@andrewserong To avoid the duplication of logic in the `if` and `else` blocks (also applies to `passwordless.jsx`) what if it was something like this?

```jsx
const recaptchaPromise = 'show' === abtest( 'userStepRecaptcha' ) ? recordGoogleRecaptchaAction( this.state.recaptchaClientId, 'calypso/signup/formSubmit' ) : Promise.resolve();

recaptchaPromise.then( token => {
	this.submit( {
		userData,
		form: formWithoutPassword,
		queryArgs: ( this.props.initialContext && this.props.initialContext.query ) || {},
		recaptchaToken: token || undefined,
	} );
} );
```

`Promise.resolve()` makes a promise that resolves immediately.

Reply:
Hey, that's a nice one @p-jackson — thanks, I'll give that a shot.

Reply:
Done! That looks a bit cleaner. Ready for another look @p-jackson